misc/python/materialize/cloudtest/app/materialize_application.py

```diff
@@ -161,3 +161,70 @@ def set_environmentd_failpoints(self, failpoints: str) -> None:
         stateful_set.env["FAILPOINTS"] = failpoints
         stateful_set.replace()
         self.wait_for_sql()
+
+    def get_k8s_value(
+        self, selector: str, json_path: str, remove_quotes: bool = True
+    ) -> str:
+        value = self.kubectl(
+            "get",
+            "pods",
+            f"--selector={selector}",
+            "-o",
+            f"jsonpath='{json_path}'",
+        )
+
+        if remove_quotes:
+            value = value.replace("'", "")
+
+        return value
+
+    def get_value_of_pod(
+        self, cluster_id: str, json_path: str, remove_quotes: bool = True
+    ) -> str:
+        return self.get_k8s_value(
+            f"cluster.environmentd.materialize.cloud/cluster-id={cluster_id}",
+            json_path,
+            remove_quotes,
+        )
+
+    def get_label_value_of_pod(
+        self, cluster_id: str, label: str, remove_quotes: bool = True
+    ) -> str:
+        return self.get_value_of_pod(
+            cluster_id, "{.items[*].metadata.labels." + label + "}", remove_quotes
+        )
+
+    def get_node_names_of_pod_by_cluster_name(self, cluster_name: str) -> List[str]:
+        cluster_id = self.load_cluster_id_from_db(cluster_name)
+        return self.get_node_names_of_pod_by_cluster_id(cluster_id)
+
+    def get_node_names_of_pod_by_cluster_id(self, cluster_id: str) -> List[str]:
+        value_string = self.get_value_of_pod(
+            cluster_id, "{.items[*].spec.nodeName}", remove_quotes=True
+        )
+        values = value_string.split(" ")
```

Comment:
If the `get_value_of_pod` function was returning a JSON you may not need to split the output like that.

Reply:
I might be wrong but it does not seem to return JSON when specifying the JSON path until the value. It returns something like `'value1 value2 value3'`. For non-leaf paths, it returns JSON.

(Though, it might be possible to formulate the query in another way that it returns a JSON object.)